### ISSUE TITLE
Crash at line 122 on ObjectMapper/Core/Mapper.swif

### ISSUE THREAD

**Issue by TofPlay** [tristanhimmelman/AlamofireObjectMapper#147]
- `carthage version`: 0.18
- Swift: 2.2
- `xcodebuild -version`:  Xcode 7.3.1 Build version 7D1014
- Are you using `--no-build`? No
- Are you using `--no-use-binaries`? No
- Are you using `--use-submodules`? No

**Cartfile**

```
github "tristanhimmelman/AlamofireObjectMapper" == 3.0.2
```

**Carthage Output**

```
*** Fetching AlamofireObjectMapper
*** Fetching ObjectMapper
*** Fetching Alamofire
*** Checking out Alamofire at "3.5.0"
*** Checking out ObjectMapper at "1.5.0"
*** Checking out AlamofireObjectMapper at "3.0.2"
*** xcodebuild output can be found in /var/folders/rw/gbr9wvs91cz91mcpgdxv4q340000gn/T/carthage-xcodebuild.jaQvVD.log
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "ObjectMapper-iOS" in ObjectMapper.xcworkspace
*** Building scheme "AlamofireObjectMapper iOS" in AlamofireObjectMapper.xcworkspace
```

For a project that I maintain in Swift 2.2 when I executes the application on the device, not on the simulator, it crash at line 122 on ObjectMapper/Core/Mapper.swif
![image](https://cloud.githubusercontent.com/assets/1082222/18472507/2bd55d64-79b8-11e6-92b3-adbe19034920.png)

Work fine if I change Cartfile like this:

```
github "Alamofire/Alamofire" == 3.4.2
github "Hearst-DD/ObjectMapper" == 1.4.0
github "tristanhimmelman/AlamofireObjectMapper" == 3.0.2
```

**Comment by csmann** [comment]
Same issue is present using Cocoapods.

**Comment by Drusy** [comment]
Same issue here using Alamofire 3.5 and ObjectMapper 1.5

**Comment by ehuynh** [comment]
the issue could be that it is being built against an older version of `ObjectMapper` as the submodules are pointing to a different version of ObjectMapper that what is being resolved as the version to use by Carthage. `AlamofireObjectMapper` is then built against one version of `ObjectMapper` and then linked to another version in your app.

**Comment by tristanhimmelman** [comment]
Is this problem still occurring. I believe it should be fixed using the latest release on swift 3.0. I will close this for now. Please add a comment if you are still having an issue and I will reopen.
